src/ui.cpp

```diff
@@ -889,8 +889,17 @@ void UpdateDialog::StateUpdateAvailable(const Appcast& info, bool installAutomat
 
     // Only show the release notes now that the layout was updated, as it may
     // take some time to load the MSIE control:
-    if ( showRelnotes )
-        ShowReleaseNotes(info);
+    if (showRelnotes)
+    {
+        // ShowReleaseNotes() has been found to generate exceptions 
+        // Uncaught exceptions will cause the CE client to terminate.
+        // Therefore, catch all exceptions, so that the client will remain active!
+        try
+        {
+            ShowReleaseNotes(info);
+        }
+        CATCH_ALL_EXCEPTIONS
+    }
```

Comment:
I have several questions, they may be out of scope, but worth mentioning...
- Were you able to recreate this exception in any way?
- How will the user experience be when this exception happens? The Client will no longer crash and that is great! Can the previous dialog still be shown so they could retry? Or it seems that the exception is thrown from `new wxActiveXContainer(m_browserParent, IID_IWebBrowser2, browser);`, would it make sense to retry that and have `try` block around it only and retry on error?
- Is there a way to propagate this error back to the Client so that it can retry maybe?
- I am not sure where the Winsparle log goes and if we should care about collecting it, but maybe we should log this event so that we know it happened?
- What are the chances that the failure (exception) will happen again on a retry?